wkwebview-engine-localhost-fix (issue: cordova-hot-code-push not working in case the application use local webserver)

salb3la opened the pull request:
With these changes the cordova-hot-code-push plugin works with the
wkwebview-engine-localhost plugin

chriswep commented:
@salb3la any idea on how to make this work with https://github.com/ionic-team/cordova-plugin-wkwebview-engine/ ?

salb3la commented:
@chriswep could you provide more information about the error?

chriswep commented:
@salb3la don't know if your PR is at all related to the issue with https://github.com/ionic-team/cordova-plugin-wkwebview-engine/ - it's described here: https://github.com/nordnet/cordova-hot-code-push/issues/264#issuecomment-289385153

Since the latest updates to the ionic-plugin you get CORS errors when accessing local files via XHR (while using the HCP plugin). Seemingly you need to drop the `file://` part in the request. But i didn't figure out how to do that in the plugin code.

salb3la commented:
@chriswep, here is the story of my pull request :)
I added these plugins to my project:
local-webserver: https://github.com/apache/cordova-plugins/tree/local-webserver
wkwebview: https://github.com/apache/cordova-plugin-wkwebview-engine
hot-code-push: https://github.com/salb3la/cordova-hot-code-push/tree/improvements

Local webserver needs to resolve CORS issues with wkwebview plugin. When I started to use the hot code push plugin, it wasnt work for me. 
The reason is that, the plugin change the start page path, but it uses file protocoll and the local webserver throw errors.
The application through the webserver can reach files from storage, but it needs to use a specific url like http://localhost:{port}/local-filesystem/var/mobile/Applications/<UUID>/.../index.html [iOS]. 
In my pull request I detect if the application uses local webserver and in that case, I cast the file protocoll path to the correct url. On my branch you could find my fix and I added another pull request, which is fix this bug https://github.com/nordnet/cordova-hot-code-push/pull/279
